CAIPs/caip-2.md

```diff
@@ -0,0 +1,111 @@
+
+## Motivation
+
+The final trigger to create this CAIP (and the CAIP process itself) was a discussion around [EIP2256] at [Ethereum-Magicians](https://ethereum-magicians.org/t/eip-2256-add-wallet-getownedtokens-json-rpc-method/3600/14).
+Independently, the [Universal Chain Registry](https://github.com/UCRegistry) was created that needs properly specified chain identifiers at its core. A [discussion about the network ID format](https://github.com/UCRegistry/registry/pull/13) brought this group together with ChainAgnostic.
+
+## Specification
+
+The blockchain ID (short "chain ID") is a string designed to uniquely identify blockchains in a developer-friendly fashion.
+
+### Syntax
+
+The `chain_id` is a case-sensitive string in the form
+
+```
+chain_id:    namespace + ":" + reference
+namespace:   [-a-z]{3,16}
+reference:   [-a-zA-Z0-9]{1,47}
+```
+
+### Semantics
+
+Each `namespace` covers a class of similar blockchains.
+Usually it describes an ecosystem or standard, such as e.g. `cosmos` or `eip155`.
+One namespace should include as many blockchains as possible.
+`reference` is a way to identify a blockchain within a given namespace.
+The semantics as well as the more granular syntax are of the reference are delegated to ecosystem specific documents, to be expected as separate CAIPs.
+
+## Rationale
+
+The goals of the general chain ID format is:
+
+- Uniqueness within the entire blockchain ecosystem
+- To some degree human readable and helps for basic debugging
+- Restricted in a way that it can be stored on chain
+- Character set basic enough to display in hardware wallets as part of a transaction content
+
+The following secondary goals can easily be archived:
+
+- Can be used unescaped in URL paths
+- Can be used as filename in a case-sensitive UNIX file system (Linux/git).
+
+Those secondary goals have been given up along the way:
+
+- Can be used as filename in a case-insensitive UNIX file system (macOS).
+- Can be used as filename in a Windows file system.
+
+## Test Cases
+
+This is a list of manually composed examples
+
+```
+# Ethereum mainnet
+eip155:1
+
+# Bitcoin mainnet (see https://github.com/bitcoin/bips/blob/master/bip-0122.mediawiki#definition-of-chain-id)
+bip122:000000000019d6689c085ae165831e93
+
+# Litecoin
+bip122:12a765e31ffd4059bada1e25190f6e98
+
+# Feathercoin (Litecoin fork)
+bip122:fdbe99b90c90bae7505796461471d89a
+
+# Cosmos Hub (Tendermint + Cosmos SDK)
+cosmos:cosmoshub-2
+cosmos:cosmoshub-3
+
+# Binance chain (Tendermint + Cosmos SDK; see https://dataseed5.defibit.io/genesis)
+cosmos:Binance-Chain-Tigris
+
+# IOV Mainnet (Tendermint + weave)
+cosmos:iov-mainnet
+
+# Lisk Mainnet (LIP-0009; see https://github.com/LiskHQ/lips/blob/master/proposals/lip-0009.md)
+lip9:9ee11e9df416b18b
+
+# Random max length (16+1+47 = 64 chars/bytes)
+blockchain1-hash:xip3343-8c3444cf8970a9e41a706fab93e7a6c4-xxxyyy
```

Comment:
Typo `blockchain1-hash` should be `blockchain1:hashed-`

Reply:
This is a dummy example of a max length chain ID. As this does not exist anyways, it can be any namespace. The important thing is length 17.

Reply:
Gotcha let's just update the example's namespace to remove ambiguity with hashed references

Reply:
I tried a few iterations, maybe `examplenamespace`

Reply:
or even better `max-namespace-16`

Reply:
Done in 74ce2ba95872c1eee65fa977961b40f935daa77c